src/modules/facility/facility.service.get-facility-by-identifier.test.ts

```diff
@@ -0,0 +1,85 @@
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const idToken = valueGenerator.string();
+
+  let acbsAuthenticationService: AcbsAuthenticationService;
+  let acbsFacilityService: AcbsFacilityService;
+  let service: FacilityService;
+
+  let acbsFacilityServiceGetFacilityByIdentifier: jest.Mock;
+
+  beforeEach(() => {
+    acbsFacilityService = new AcbsFacilityService(null, null);
+
+    acbsFacilityServiceGetFacilityByIdentifier = jest.fn();
+    acbsFacilityService.getFacilityByIdentifier = acbsFacilityServiceGetFacilityByIdentifier;
+
+    const mockAcbsAuthenticationService = getMockAcbsAuthenticationService();
+    acbsAuthenticationService = mockAcbsAuthenticationService.service;
+    const acbsAuthenticationServiceGetIdToken = mockAcbsAuthenticationService.getIdToken;
+    when(acbsAuthenticationServiceGetIdToken).calledWith().mockResolvedValueOnce(idToken);
+
+    service = new FacilityService(acbsAuthenticationService, acbsFacilityService, dateStringTransformations);
+  });
+
+  describe('getFacilityByIdentifier', () => {
+    const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+    const facilityIdentifier = valueGenerator.stringOfNumericCharacters({ length: 10 });
+    const { facilitiesInAcbs, facilities } = new GetFacilityGenerator(valueGenerator, dateStringTransformations).generate({
+      numberToGenerate: 1,
+      facilityIdentifier,
+      portfolioIdentifier,
+    });
+
+    it('returns a transformation of the facility from ACBS when UserDefinedDate1 IS NOT null', async () => {
+      const userDefinedDate1InAcbs = '2023-04-11T00:00:00Z';
+      const expectedIssueDate = '2023-04-11';
+      const facilityInAcbs: AcbsGetFacilityResponseDto = {
+        ...facilitiesInAcbs[0],
+        UserDefinedDate1: userDefinedDate1InAcbs,
+      };
+      const expectedFacility: GetFacilityByIdentifierResponseDto = {
+        ...facilities[0],
+        issueDate: expectedIssueDate,
+      };
+      when(acbsFacilityServiceGetFacilityByIdentifier).calledWith(facilityIdentifier, idToken).mockResolvedValueOnce(facilityInAcbs);
+
+      const facility = await service.getFacilityByIdentifier(facilityIdentifier);
+
+      expect(facility).toStrictEqual(expectedFacility);
+    });
+
+    it('returns a transformation of the external ratings from ACBS when OfficerRiskDate IS null', async () => {
+      const facilityInAcbs: AcbsGetFacilityResponseDto = {
+        ...facilitiesInAcbs[0],
+        UserDefinedDate1: null,
+      };
+      const expectedFacility: GetFacilityByIdentifierResponseDto = {
+        ...facilities[0],
+        issueDate: null,
+      };
+      when(acbsFacilityServiceGetFacilityByIdentifier).calledWith(facilityIdentifier, idToken).mockResolvedValueOnce(facilityInAcbs);
+
+      const facility = await service.getFacilityByIdentifier(facilityIdentifier);
+
+      expect(facility).toStrictEqual(expectedFacility);
+    });
```

Comment:
These tests are a bit strange to me - why do we only test when `UserDefinedDate1` is not null and when `OfficerRiskDate` is null? What about when `UserDefinedDate1` is null, when `OfficerRiskDate` is not null, and what about `ExpirationDate`, `UserDefinedDate2`, and `OriginalEffectiveDate`? Let me know what you think on those, I think we should test any that we think could be null

It feels like we should definitely test when `CompBalPctReserve`, and `CompBalPctAmount` are null because they have defaults

Reply:
Good points. The tests actually currently test when `UserDefinedDate1` is not null and also when it is null (there's just a mistake in the test description of the latter test as I copied the tests from another endpoint as a starting point and evidently forgot to change this description). 

In the Mulesoft code, there is only a null check for `UserDefinedDate1`, which made me think that the other dates could not be null, but looking at the ACBS docs this does not appear to the case, so I've added tests for all the other dates as well.